sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -289,20 +296,41 @@ ItemBuilder addIfNotDefault(
     ItemBuilder add(String key, Class<?> value);
 
     /**
+     * Register the given class display metadata. The metadata item will be registered with type
+     * {@link DisplayData.Type#JAVA_CLASS}, and is identified by the specified key and namespace
+     * from the current transform or component.
+     */
+    ItemBuilder add(String key, JavaClass value);
+
+    /**
      * Register the given class display data if the value is not null.
      *
      * @see DisplayData.Builder#add(String, Class)
      */
     ItemBuilder addIfNotNull(String key, @Nullable Class<?> value);
 
     /**
+     * Register the given class display data if the value is not null.
+     *
+     * @see DisplayData.Builder#add(String, JavaClass)
+     */
+    ItemBuilder addIfNotNull(String key, @Nullable JavaClass value);
+
+    /**
      * Register the given class display data if the value is different than the specified default.
      *
      * @see DisplayData.Builder#add(String, Class)
      */
     ItemBuilder addIfNotDefault(
         String key, @Nullable Class<?> value, @Nullable Class<?> defaultValue);
 
+    /**
+     * Register the given class display data if the value is different than the specified default.
+     *
+     * @see DisplayData.Builder#add(String, JavaClass)
+     */
+    ItemBuilder addIfNotDefault(
+        String key, @Nullable JavaClass value, @Nullable JavaClass defaultValue);
```

Comment:
Maybe call this `DisplayJavaClass`, `DisplayClass` or `ClassForDisplay` to emphasize the relation to display data?